src/api/api.cpp

```diff
@@ -138,6 +141,24 @@ void quantize_int8_wrap(program& prog, const target& t, quantize_int8_options& o
     migraphx::quantize_int8(prog, t, options.calibration, options.op_names);
 }
 
+operation create_op(const char* name, const char* attributes)
+{
+    value v = value::object{};
+    if(attributes != nullptr)
+    {
+        v = from_json_string(convert_to_json(std::string(attributes)));
+    }
+    auto op = make_op(name, v);
+
+    return op;
+}
+
+const char* op_name(const operation& op)
+{
+    const auto name = std::make_unique<std::string>(op.name());
+    return name->c_str();
```

Comment:
Why cant this be `op.name().c_str()`?

Reply:
Actually this wont work as the string will be deallocated causing a dangling reference. We need to update the operation interface to return a string_view. Let me look into a PR to update that.

Reply:
Actually, `string_view` wont work either since its not guaranteed to be a null terminated string. I think the API needs to be change to fill a buffer. api.py can automatically do this, but a wrapper needs to be added. You can do something similar to the [`vector_c_wrap`](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/tools/api.py#L700) for strings:

```python
@cwrap('std::string')
def string_c_wrap(p):
    t = 'const char*'
    if p.returns:
        if p.type.is_reference():
            p.add_param(t.add_pointer())
            p.bad_param('${name} == nullptr', 'Null pointer')
            p.cpp_write = '${type}(${name})'
            p.write = [
                '*${name} = ${result}.c_str()'
            ]
        else:
            p.add_param(t)
            p.bad_param('${name} == nullptr', 'Null pointer')
            p.cpp_write = '${type}(${name})'
            p.write = [
                'std::copy(${result}.begin(), ${result}.end(), ${name})'
            ]
    else:
        p.add_param(t)
        p.bad_param('${name} == nullptr', 'Null pointer')
        p.read = '${type}(${name})'
```

Reply:
Thanks for the help. I made changes accordingly.